### PR TITLE
allow the use of a port number in the image's repo url

### PR DESCRIPTION
**Pull request by chilicheech**
This is useful for enterprises that have their own private docker registries using a non-default port. For example, if my registry lives at `my.awesome.docker.registry:8081` and my image is at `ubuntu/xenial` I would then be able to set my image to `my.awesome.docker.registry:8081/ubuntu/ubuntu:xenial` in the `.kitchen.yaml`.

This makes it so you must use a tag if you want to use a port. Otherwise anything after the colon gets interpreted as the tag. This also preserves the default behavior of defaulting to tag `latest` when a port is not passed. Here are the valid uses with this method:

```
image: ubuntu/ubuntu
image: ubuntu/ubuntu:latest
image: my.awesome.docker.registry:8081/ubuntu/ubuntu:latest
```
in other words:
```
image: repo_without_port
image: repo_without_port:tag
image: repo_with_port:tag
```

### CODE REVIEW

**Comment by someara** [comment]
released in 2.0.10

**Comment by chilicheech** [comment]
Holy smokes that was fast!
